src/transaction/transaction_context.cpp

```diff
@@ -46,6 +53,10 @@ void TransactionContext::Commit() {
 	auto transaction = std::move(current_transaction);
 	ClearTransaction();
 	auto error = transaction->Commit();
+	// Notify any registered state of transaction commit
+	for (auto const &s : context.registered_state) {
+		s.second->TransactionCommit(*transaction, context);
```

Comment:
In case of `error.HasError()` should we emit the TransactionRollback event here?

Reply:
Makes sense, done